overlord/snapstate/snapstate_update_test.go

```diff
@@ -16121,6 +16121,462 @@ func (s *snapmgrTestSuite) testUpdateWithComponentsRunThrough(c *C, opts updateW
 	}
 }
 
+func (s *snapmgrTestSuite) TestUpdateWithComponentsRunThroughShareComponents(c *C) {
+	const undo = false
+	s.testUpdateWithComponentsRunThroughShareComponents(c, undo)
+}
+
+func (s *snapmgrTestSuite) TestUpdateWithComponentsRunThroughShareComponentsUndo(c *C) {
+	const undo = true
+	s.testUpdateWithComponentsRunThroughShareComponents(c, undo)
+}
+
+func (s *snapmgrTestSuite) testUpdateWithComponentsRunThroughShareComponents(c *C, undo bool) {
+	s.enableRefreshAppAwarenessUX()
+
+	const (
+		snapName = "some-snap"
+		snapID   = "some-snap-id"
+	)
+
+	channel := "channel-for-components"
+	currentSnapRev := snap.R(7)
+
+	newSnapRev := snap.R(11)
+
+	components := []string{"kernel-modules-component", "standard-component"}
```

Comment:
This should be a snap of `TypeKernel` if using `kernel-modules` components

Reply:
Thanks, done